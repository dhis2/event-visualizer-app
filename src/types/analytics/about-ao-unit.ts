import type { FC } from 'react'

type AboutAOUnitProps = {
    id: string
    type: string
    renderId?: number
}

export type AboutAOUnit = FC<AboutAOUnitProps>
