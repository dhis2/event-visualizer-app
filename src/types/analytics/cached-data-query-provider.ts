import { Query } from '@dhis2/app-service-data'
import { FC, ReactNode } from 'react'

type CachedDataQueryProviderProps = {
    children: ReactNode
    query: Query
    dataTransformation?: (data: unknown) => unknown
}

export type CachedDataQueryProvider = FC<CachedDataQueryProviderProps>
export type useCachedDataQuery<T> = () => T
