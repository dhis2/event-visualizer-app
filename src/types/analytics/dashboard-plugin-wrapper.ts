import type { FC, ReactNode } from 'react'

type RenderProp<P> = (props: P) => ReactNode

type DashboardPluginWrapperProps<P> = {
    cacheId: string
    children: RenderProp<P>
    isParentCached?: boolean
    onInstallationStatusChange?: () => void
}

export type DashboardPluginWrapper<P> = FC<DashboardPluginWrapperProps<P>>
