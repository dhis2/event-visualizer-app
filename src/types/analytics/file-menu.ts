import type { FC } from 'react'
import type {
    CurrentUser,
    CurrentVisualization,
    EventVisualizationType,
    VisualizationType,
} from '@types'

type FileType = 'visualization' | 'eventVisualization' | 'eventReport' | 'map'

type FileMenuRenamePayload = {
    name: string
    description: string
}

// TODO check if any of these should be required
type FileMenuProps = {
    currentUser?: CurrentUser
    defaultFilterVisType?: string
    fileObject?: CurrentVisualization
    fileType?: FileType
    filterVisTypes?: Array<{
        type: EventVisualizationType | VisualizationType | string
    }>
    onDelete?: () => void
    onError?: (error: string) => void
    onNew?: () => void
    onOpen?: (id: string) => void
    onRename?: ({ name, description }: FileMenuRenamePayload) => void
    onSave?: () => void
    onSaveAs?: ({ name, description }: FileMenuRenamePayload) => void
    onShare?: () => void
    onTranslate?: () => void
}

export type FileMenu = FC<FileMenuProps>
