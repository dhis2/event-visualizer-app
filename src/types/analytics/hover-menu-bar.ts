import type { FC, ReactNode } from 'react'

type HoverMenuBarProps = {
    children: ReactNode
    dataTest?: string
}

export type HoverMenuBar = FC<HoverMenuBarProps>
