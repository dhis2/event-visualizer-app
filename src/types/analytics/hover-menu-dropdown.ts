import type { FC, ReactNode } from 'react'

type HoverMenuDropdownProps = {
    children: ReactNode
    className?: string
    dataTest?: string
    disabled?: boolean
    label?: ReactNode
}

export type HoverMenuDropdown = FC<HoverMenuDropdownProps>
