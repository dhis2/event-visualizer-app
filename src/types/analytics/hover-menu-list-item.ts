import type { FC, ReactNode } from 'react'

type HoverMenuListItemProps = {
    children?: ReactNode
    className?: string
    dataTest?: string
    destructive?: boolean
    disabled?: boolean
    icon?: ReactNode
    label?: ReactNode
    onClick?: () => void
}

export type HoverMenuListItem = FC<HoverMenuListItemProps>
