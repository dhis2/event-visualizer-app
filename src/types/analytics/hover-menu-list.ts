import type { FC, ReactNode } from 'react'

type HoverMenuListProps = {
    children: ReactNode
    className?: string
    dataTest?: string
    dense?: boolean
    maxHeight?: string
    maxWidth?: string
}

export type HoverMenuList = FC<HoverMenuListProps>
