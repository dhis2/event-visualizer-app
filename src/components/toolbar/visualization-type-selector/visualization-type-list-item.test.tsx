import { VIS_TYPE_LINE_LIST } from '@dhis2/analytics'
import { render } from '@testing-library/react'
import React from 'react'
import { vi } from 'vitest'
import VisualizationTypeListItem from './visualization-type-list-item'

test('VisualizationTypeListItem renders Line list item', () => {
    const props = {
        iconType: VIS_TYPE_LINE_LIST,
        label: 'Line list',
        description: 'A visualization that displays data in a line list format',
        disabled: false,
        isSelected: false,
        onClick: vi.fn(),
    }

    const { container } = render(<VisualizationTypeListItem {...props} />)
    expect(container).toMatchSnapshot()
})
