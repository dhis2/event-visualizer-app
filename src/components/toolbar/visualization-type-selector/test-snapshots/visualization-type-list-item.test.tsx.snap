// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VisualizationTypeListItem renders Line list item 1`] = `
<div>
  <div
    class="_listItem_51a851"
  >
    <div
      class="_listItemIcon_51a851"
    >
      <div
        style="width: 48px; height: 48px;"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M13 4h7v2h-7V4zm-2 2V4H4v2h7zM4 8h7v5H4V8zm9 5V8h7v5h-7zM2 6V2h20v20H2V6zm2 9v5h7v-5H4zm9 0v5h7v-5h-7z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="_listItemText_51a851"
    >
      <p
        class="_listItemName_51a851"
      >
        Line list
      </p>
      <p
        class="_listItemDescription_51a851"
      >
        A visualization that displays data in a line list format
      </p>
    </div>
  </div>
</div>
`;
